### DISCUSSION TITLE
Android version and NNUE.

### DISCUSSION THREAD

**Discussion by GustavoConde22** [official-stockfish/Stockfish#3778] [Q&A] [answered]
in the android version of stockfish 14.1, what is the difference between armv7 and armv7 neon? and is the nnue mode included in the android version?

**Comment by vondele** [comment, marked as answer]
armv7-neon is a binary that is using neon instructions, which are available in almost all armv7 devices. The other binary doesn't use these instructions for the few devices out there that do not support neon, and is thus slower. NNUE is automatically used with 14.1

**Comment by cemengorbunkov** [comment]
Ребята здравствуйте! Подскажите пожалуйста у меня перестали работать(На "Android"(v.4.1.2)-armv7 neon!)последние версии "Stockfish" которые по 110mb.(В предварительной версии двигателя - "dev"!),...все прежние версии "Stockfish"(На "Android"!)включая версию "17.1" (77mb.)работают нормально,хорошо! Почему не работают последние версии двигателя "Stockfish dev" для "Android"(v.4.1.2),может быть не хватает "Оперативной(доступной)памяти" телефона(1000-1200mb.)?! Спасибо,спасибо большое!

**Reply by vondele** [reply]
Hello guys! Please tell me, the latest versions of Stockfish, which are 110 MB, have stopped working for me (on Android (v. 4.1.2) - armv7 neon). (In the preliminary version of the engine - "dev"!),... All previous versions of Stockfish (on Android!), including version "17.1" (77 MB), work fine, well! Why aren't the latest versions of the "Stockfish dev" engine for Android (v. 4.1.2) working? Maybe the phone's "RAM (available) memory" (1000-1200 MB) is not enough?! Thank you, thank you very much!

**Reply by vondele** [reply]
in principle a SF process should use less than 1GB of memory, even with the lastest network. Do you see any error output?

**Reply by vondele** [reply]
Could you try the very latest commit?

**Reply by cemengorbunkov** [reply]
...Да,вот эта версия движка "Stockfish dev-20260128-253aaefb" тоже не работает на "Android"(v.4.1.2)! У меня на телефоне стоит Шахматная программка "DroidFish" (v.1.90) (со встроенным движком "Stockfish 16.1"),всё работает в порядке,а когда возникает проблема с двигателем,программа показывает стандартное сообщение - "Ошибка движка: Работа движка остановлена",точно так же как и с этой версией ""Stockfish dev"! Ребята,я устанавливал версии движка "Stockfish dev" ранее,когда они были по"77mb."и все они отлично работают,перестали работать версии по"110mb."как "armv7 neon" так же и "armv7"! Пожалуйста,если у Вас будут какие-то изменения или сообщения,сообщите пожалуйста,ок?! Всего Вам хорошего,спасибо!

**Reply by ppigazzini** [reply]
> Yes, this version of the engine, “Stockfish dev-20260128-253aaefb,” also does not work on Android (v.4.1.2)! I have the chess program “DroidFish” (v.1.90) (with the built-in engine “Stockfish 16.1”) on my phone, everything works fine, but when there is a problem with the engine, the program displays a standard message - “Engine error: Engine stopped working”, just like with this version of “Stockfish dev”! Guys, I installed versions of the “Stockfish dev” engine earlier, when they were “77mb,” and they all work great, but the ‘110mb’ versions stopped working, both “armv7 neon” and “armv7”! Please, if you have any changes or messages, let me know, okay? All the best, thank you!

I tried bringing a very old ARMv7-neon phone of mine back to life to test SF, termux apk wouldn’t even start downloading, play store didn't start. After a few attempts, I just gave up.

**Reply by Craftyawesome** [reply]
Try resetting SF's settings and reducing hash size. I tried on 1 and 2 GiB devices and I can only get an engine terminated error if I had a wrong settings like wrong net name or I set hash too high with .unsafehash.

**Reply by anematode** [reply]
@cemengorbunkov Hello good sir! Could you try building the engine in this PR and see if it fixes your issue: https://github.com/official-stockfish/Stockfish/pull/6574

**Reply by ppigazzini** [reply]
GitHub Action has already built the engines, scroll down to the section "Artifacts", click to download:
https://github.com/official-stockfish/Stockfish/actions/runs/21506001267

<img width="916" height="432" alt="image" src="https://github.com/user-attachments/assets/5d120e1f-cc4e-436c-a2db-8bdf33285bf1" />

**Comment by ppigazzini** [comment]
A memory constrained armv7 container needs 280m to bench without a OOM kill:

Setup containers (install docker if necessary, see https://docs.docker.com/engine/install/ubuntu/)
```bash
#!/usr/bin/env bash
set -e

echo "=== Installing QEMU and enabling multi-arch ==="
sudo apt update
sudo apt install -y qemu-user-static binfmt-support

echo "=== Setting up Docker buildx ==="
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
docker buildx create --use --name multiarch || true
docker buildx inspect --bootstrap

echo "=== Cloning Stockfish (only if missing) ==="
if [ ! -d "Stockfish" ]; then
  git clone https://github.com/official-stockfish/Stockfish.git
else
  echo "Stockfish folder already exists — skipping clone"
fi

echo "=== Creating build_sf.sh inside Stockfish/src ==="
cat > Stockfish/src/build_sf.sh << 'EOF'
#!/bin/sh
cd /Stockfish/src
make clean
make -j build
EOF

echo "=== Creating bench_sf.sh inside Stockfish/src ==="
cat > Stockfish/src/bench_sf.sh << 'EOF'
#!/bin/sh
cd /Stockfish/src
./stockfish bench
EOF

echo "=== Building ARMv7 builder image (inline Dockerfile) ==="
docker buildx build \
  --platform linux/arm/v7 \
  --load \
  -t sf_builder \
  - << 'EOF'
FROM arm32v7/alpine
WORKDIR /Stockfish
RUN apk update && apk add g++ make  musl-dev
CMD ["sh", "src/build_sf.sh"]
EOF

echo "=== Building ARMv7 bencher image (inline Dockerfile) ==="
docker buildx build \
  --platform linux/arm/v7 \
  --load \
  -t sf_bencher \
  - << 'EOF'
FROM arm32v7/alpine
WORKDIR /Stockfish
RUN apk update && apk add libstdc++
CMD ["sh", "src/bench_sf.sh"]
EOF

echo "=== Running builder (compiling Stockfish ARMv7) ==="
docker run \
  --platform linux/arm/v7 \
  -v "$(pwd)/Stockfish:/Stockfish" \
  sf_builder

echo "=== DONE ==="
echo "Stockfish binary is at: $(pwd)/Stockfish/src/stockfish"
```
Memory constrained bench, eg: `bash bench.sh 280m`:
```bash
#!/usr/bin/env bash
set -e

MEMORY="${1:-512m}"

echo "=== Running bencher (testing Stockfish under RAM limit: $MEMORY) ==="
docker run \
  --platform linux/arm/v7 \
  --memory="$MEMORY" \
  --memory-swap="$MEMORY" \
  -v "$(pwd)/Stockfish:/Stockfish" \
  sf_bencher
```
